### PR TITLE
Fix updating session list to main server

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java [hunk 1]
```diff
@@ -81,10 +81,6 @@ public void leaveSession(Player player) {
 
         PlayerLeftSessionMessage m = new PlayerLeftSessionMessage(player.getUsername(), team.getTeamColour());
         room.getAllPlayers().stream().map(server::getConnectionFromPlayer).forEach(c -> c.send(m));
-
-        if (room.getOccupancy() <= 0) {
-            controller.destroySession(session);
-        }
     }
 
     public void startGame() {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java:86]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 2]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
...
 
+    public void increaseOccupancy(String roomName) {
+        try {
+            mainServerForGameServer.increaseSessionOccupancy(gameServerForMainServer, roomName);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
+        }
+    }
+
+    public void decreaseOccupancy(String roomName) {
```
Comment (TehMessiah) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 3]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
         }
     }
 
+    public void increaseOccupancy(String roomName) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java:124]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 4]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
...
+
+    public void decreaseOccupancy(String roomName) {
+        try {
+            mainServerForGameServer.decreaseSessionOccupancy(gameServerForMainServer, roomName);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
+        }
+    }
+
+    public void changeHost(String roomName, String host) {
```
Comment (TehMessiah) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 12 issues
* MAJOR 1 major
* MINOR 10 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [SessionWrapper.java#L15](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L15): Remove this unused "controller" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [SessionWrapper.java#L12](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L12): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [SessionWrapper.java#L18](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L18): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [SessionWrapper.java#L24](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L24): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [SessionWrapper.java#L75](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L75): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerController.java#L116](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java#L116): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerController.java#L120](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java#L120): Define a constant instead of duplicating this literal "An error occurred while reporting a destroyed session." 4 times. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1192)
1. INFO [SessionController.java#L77](https://github.com/PTS3-S34A/Game-Server/blob/381e3c10f0c1113996fed8f9cb9644a1321e3108/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java#L77): Complete the task associated to this TODO comment. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1135)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 5]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
...
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
+        }
+    }
+
+    public void decreaseOccupancy(String roomName) {
+        try {
+            mainServerForGameServer.decreaseSessionOccupancy(gameServerForMainServer, roomName);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
```
Comment (Laugslander) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java]: Logger message aanpassen voor deze methode.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 6]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
         }
     }
 
+    public void increaseOccupancy(String roomName) {
+        try {
+            mainServerForGameServer.increaseSessionOccupancy(gameServerForMainServer, roomName);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
```
Comment (Laugslander) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java]: Logger message aanpassen voor deze methode.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 7]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
...
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
+        }
+    }
+
+    public void changeHost(String roomName, String host) {
+        try {
+            mainServerForGameServer.hostChanged(gameServerForMainServer, roomName, host);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting a destroyed session.", e);
```
Comment (Laugslander) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java]: Logger message aanpassen voor deze methode.

**Review by Laugslander — CHANGES_REQUESTED** [review]

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 8]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
...
 
+    public void increaseOccupancy(String roomName) {
+        try {
+            mainServerForGameServer.increaseSessionOccupancy(gameServerForMainServer, roomName);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting increased occupancy for a session.", e);
+        }
+    }
+
+    public void decreaseOccupancy(String roomName) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java:132]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java [hunk 9]
```diff
@@ -123,4 +121,28 @@ public void sessionDestroyed(String roomName) {
...
+        try {
+            mainServerForGameServer.decreaseSessionOccupancy(gameServerForMainServer, roomName);
+        } catch (RemoteException e) {
+            LOGGER.log(Level.SEVERE, "An error occurred while reporting decreased occupancy for a session.", e);
+        }
+    }
+
+    public void changeHost(String roomName, String host) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java:140]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 11 issues
* MAJOR 1 major
* MINOR 9 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [SessionWrapper.java#L15](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L15): Remove this unused "controller" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [SessionWrapper.java#L12](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L12): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [SessionWrapper.java#L18](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L18): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [SessionWrapper.java#L24](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L24): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [SessionWrapper.java#L75](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionWrapper.java#L75): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GameServerController.java#L116](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/rmi/GameServerController.java#L116): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. INFO [SessionController.java#L77](https://github.com/PTS3-S34A/Game-Server/blob/c2d174aedb1e31f33cd141e31c54ea32a841015d/Soccar [Game-Server]/src/nl/soccar/gameserver/SessionController.java#L77): Complete the task associated to this TODO comment. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1135)

**Review by Laugslander — APPROVED** [review]

**Pull request merged**